### PR TITLE
Add support for RWPM .audiobook files

### PR DESCRIPTION
**Pull request by vbessonov**
## Description

This PR adds media types for supporting RWPM `.audiobook` files. You can find more information in [SIMPLY-3183](https://jira.nypl.org/browse/SIMPLY-3183). 

This PR depends on [PR # 1203 in server_core](https://github.com/NYPL-Simplified/server_core/pull/1203).

## Checklist:

- [x] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts.py [hunk 1]
```diff
@@ -1309,6 +1309,14 @@ def arg_parser(cls, _db):
             help=u"Show what would be imported, but don't actually do the import.",
             action='store_true',
         )
+        parser.add_argument(
+            '--default-medium-type',
+            help=u'Default medium type used in the case when it\'s not explicitly specified in a metadata file. '
+                 u'Valid values are: Book, Audio.',
```
Comment (leonardr) [inline scripts.py:1315]: Substitute EditionConstants.FULFILLABLE_MEDIA both here and in `choices` below.

File: scripts.py [hunk 2]
```diff
@@ -1323,16 +1331,33 @@ def do_run(self, cmd_args=None):
...
+            collection_name,
+            collection_type,
+            data_source_name,
+            metadata_file,
+            metadata_format,
+            cover_directory,
+            ebook_directory,
+            rights_uri,
+            dry_run,
+            default_medium_type
```
Comment (leonardr) [inline scripts.py:1346]: It's up to you, but at this point when I'm adding yet another argument to a method that takes a lot of arguments, I usually switch over to passing in keyword arguments. This reduces the risk that I'll mix up the order of two arguments and create a difficult-to-find bug.

File: tests/test_scripts.py [hunk 3]
```diff
@@ -877,7 +877,8 @@ def run_with_arguments(self, *args):
                 "--cover-directory=covers",
                 "--ebook-directory=ebooks",
                 "--rights-uri=rights",
-                "--dry-run"
+                "--dry-run",
+                "--default-medium-type=Audiobook"
```
Comment (leonardr) [inline tests/test_scripts.py:881]: This should be `Audio`, not `Audiobook`. This is what's causing the test failure -- the script is exiting early due to invalid command-line arguments.

**Review by leonardr — CHANGES_REQUESTED** [review]
This looks good once the test failure is fixed and the list of allowable media is changed to use the list defined in EditionConstants.

File: scripts.py (hunk 1 above)
Comment (vbessonov) [reply scripts.py]: I changed the code accordingly. You also wanted to replace `u'Valid values are: Book, Audio.'` with `u'Valid values are: EditionConstants.FULFILLABLE_MEDIA.'`, right?

File: scripts.py (hunk 2 above)
Comment (vbessonov) [reply scripts.py]: It's a good advice, changed the code accordingly

File: tests/test_scripts.py (hunk 3 above)
Comment (vbessonov) [reply tests/test_scripts.py]: Sorry, didn't notice it at first, strangely enough the tests worked locally (probably had some other changes). I replaced `Audiobook` with `Audio`

File: scripts.py (hunk 1 above)
Comment (leonardr) [reply scripts.py]: Sorry, I should have been more clear. I meant something like this:
```
"Valid values are: {}".format(", ".join(EditionConstants.FULFILLABLE_MEDIA))
```

Comment (vbessonov) [reply scripts.py]: My bad, fixed it

File: scripts.py [hunk 4]
```diff
@@ -1312,9 +1312,9 @@ def arg_parser(cls, _db):
         parser.add_argument(
             '--default-medium-type',
             help=u'Default medium type used in the case when it\'s not explicitly specified in a metadata file. '
-                 u'Valid values are: Book, Audio.',
+                 u'Valid values are: EditionConstants.FULFILLABLE_MEDIA.',
```
Comment (leonardr) [inline scripts.py:1315]: I think you changed the default_medium_type but not this piece, which is the confusing part.

Comment (vbessonov) [reply scripts.py]: That's weird, I must've forgotten to push

**Review by leonardr — APPROVED** [review]

**Pull request merged**